eth/eventhandler/handlers.go

```diff
@@ -54,13 +54,16 @@ func (eh *EventHandler) handleOperatorAdded(txn basedb.Txn, event *contract.Cont
 
 	// throw an error if there is an existing operator with the same public key and different operator id
 	operatorData := eh.operatorDataStore.GetOperatorData()
-	if operatorData.ID != 0 && bytes.Equal(operatorData.PublicKey, event.PublicKey) && operatorData.ID != event.OperatorId {
+	_, found, err := eh.nodeStorage.GetOperatorDataByPubKey(txn, event.PublicKey)
+	if err != nil {
+		return fmt.Errorf("could not get operator data by public key: %w", err)
+	}
+	if found || operatorData.ID != 0 && bytes.Equal(operatorData.PublicKey, event.PublicKey) && operatorData.ID != event.OperatorId {
```

Comment:
I think the point was to check the operator data from the storage against the event operator data. 
`operatorData` is our *own* operator. we want to compare the event to the operator found in the DB/

Reply:
Agree. Fixed